http-server-netty/src/main/java/io/micronaut/http/server/netty/multipart/MultipartBodyArgumentBinder.java

```diff
@@ -76,88 +79,97 @@ public Argument<MultipartBody> argumentType() {
 
     @Override
     public BindingResult<MultipartBody> bind(ArgumentConversionContext<MultipartBody> context, HttpRequest<?> source) {
-        if (source instanceof NettyHttpRequest) {
-            NettyHttpRequest nettyHttpRequest = (NettyHttpRequest) source;
-            io.netty.handler.codec.http.HttpRequest nativeRequest = nettyHttpRequest.getNativeRequest();
-            if (nativeRequest instanceof StreamedHttpRequest) {
-                HttpContentProcessor processor = beanLocator.findBean(HttpContentSubscriberFactory.class,
-                        new ConsumesMediaTypeQualifier<>(MediaType.MULTIPART_FORM_DATA_TYPE))
-                        .map(factory -> factory.build(nettyHttpRequest))
-                        .orElse(new DefaultHttpContentProcessor(nettyHttpRequest, httpServerConfiguration.get()));
-
-                //noinspection unchecked
-                return () -> Optional.of(subscriber -> HttpContentProcessorAsReactiveProcessor.<HttpData>asPublisher(processor.resultType(context.getArgument()), nettyHttpRequest).subscribe(new CompletionAwareSubscriber<>() {
-
-                    Subscription s;
-                    AtomicLong partsRequested = new AtomicLong(0);
-
-                    @Override
-                    protected void doOnSubscribe(Subscription subscription) {
-                        this.s = subscription;
-                        subscriber.onSubscribe(new Subscription() {
-
-                            @Override
-                            public void request(long n) {
-                                if (partsRequested.getAndUpdate(prev -> prev + n) == 0) {
-                                    s.request(n);
-                                }
-                            }
+        if (source instanceof NettyHttpRequest<?> nhr) {
+            HttpContentProcessor processor = beanLocator.findBean(HttpContentSubscriberFactory.class,
+                    new ConsumesMediaTypeQualifier<>(MediaType.MULTIPART_FORM_DATA_TYPE))
+                .map(factory -> factory.build(nhr))
+                .orElse(new DefaultHttpContentProcessor(nhr, httpServerConfiguration.get()));
+            MultiObjectBody multiObjectBody;
+            try {
+                multiObjectBody = nhr.rootBody()
+                    .processMulti(processor);
+            } catch (Throwable e) {
+                throw new RuntimeException(e);
+            }
+            //noinspection RedundantCast,unchecked
+            return () -> Optional.of(subscriber -> ((Publisher<HttpData>) multiObjectBody.asPublisher()).subscribe(new CompletionAwareSubscriber<>() {
```

Comment:
My PR doesn't use a custom subscriber as it isn't needed just a simple filter/map